infra-as-code/state/pull_state.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+set -e
+echo "Trying to import Terraform State for $ENVIRONMENT"
+echo "### Job Results "
+
+IFS="|" read -ra tf_state_files <<<"$TF_STATE_FILES"
+
+tfstate_bucket=$(twilio api:serverless:v1:services:fetch --sid tfstate -o json | jq -c '.[].domainBase // empty' | sed 's/"//g')
+
+if [ -n "$tfstate_bucket" ]; then
```

Comment:
i realize it complicates things a little more but do you think we could store and retrieve these as protected assets?

Reply:
I tried to do that, but I didn't find a way to download the asset.
Here, it says that only Twilio can download a protected asset.
<img width="667" alt="image" src="https://github.com/twilio-professional-services/flex-project-template/assets/27789416/c5dfd769-24af-4b95-a1e7-b34332191293">
I tried to do curl -u "$TWILIO_ACCOUNT_SID:$TWILIO_AUTH_TOKEN" -o - $url but it didn't work too :( 
Do you know some other way to do that? Thank you.

Reply:
followed up directly.

Reply:
FIxed. Bui we will need the TWILIO_AUTH_TOKEN now!

Reply:
Is it possible to do this without TWILIO_AUTH_TOKEN? I'm not sure we want to promote using the auth token in CI.

Reply:
What if we use a private asset, but use a public function to access it? The public function could accept the API key/secret and validate it before returning the asset data.

Reply:
Here is an example of exactly this! 😄 https://github.com/twilio-professional-services/flex-project-template/blob/main/serverless-schedule-manager/functions/admin/fetch-config.js